restli-server/src/main/java/com/linkedin/restli/server/BaseRestLiServer.java

```diff
@@ -233,4 +237,69 @@ private RestLiArgumentBuilder lookupArgumentBuilder(ResourceMethodDescriptor met
     }
     return argumentBuilder;
   }
+
+  /**
+   * Method is used to validate the request's batch size is under
+   * the allowed max batch size which is defined in the server resource.
+   *
+   * @throws RestLiServiceException if request's batch size is larger than the allowed max batch size.
+   */
+  private void validateMaxBatchSize(RestLiRequestData requestData, ResourceMethodDescriptor method,
+      ServerResourceContext context) throws RestLiServiceException
+  {
+    MaxBatchSizeSchema maxBatchSizeAnnotation = method.getMaxBatchSize();
```

Comment:
nit, I think we can check method first, and return early if the method does not have anything to do with batches.

Reply:
MaxBatchSize annotation is only allowed to add on batch methods. We have such guard in annotation reader. If maxBatchSizeAnnotation == null, it is either a non-batch method or batch method without max batch size annotation. In another word, it includes check the method is batch method or not. We do not need to specify check method itself.  I do not see much value provided if we specifically check the method type first here.